### PR TITLE
WIP: Add a preference to disable Google Cast

### PR DESCRIPTION
**Pull request by jvoisin**
This commit adds a preference to disable/enable
support for the Google Cast protocol.

It's disabled by default, since I'm not comfortable
with the default airsonic setup doing requests
to Google stuff.

### CODE REVIEW

**Comment by muff1nman** [comment]
Should this not be wrapped up with the other changes in #993 ? Also I don't think having this disabled by default is good for the first run of this. Maybe in a subsequent release this can be changed.

**Comment by jvoisin** [comment]
I think that doing it step by step is a better approach than trying to do everything at once.

File: airsonic-main/src/main/java/org/airsonic/player/service/SettingsService.java [hunk 1]
```diff
@@ -123,6 +123,8 @@
     private static final String KEY_RECAPTCHA_SITE_KEY = "ReCaptchaSiteKey";
     private static final String KEY_RECAPTCHA_SECRET_KEY = "ReCaptchaSecretKey";
 
+    private static final String KEY_GOOGLE_CAST_ENABLED = "GoogleCast";
```
Comment (fxthomas) [inline airsonic-main/src/main/java/org/airsonic/player/service/SettingsService.java]: Minor nit: I would rather call it `GoogleCastEnabled` or something to that effect.

File: airsonic-main/src/main/webapp/WEB-INF/jsp/playQueue.jsp [hunk 2]
```diff
@@ -687,13 +706,13 @@
         currentStreamUrl = song.streamUrl;
         updateCurrentImage();
 
-        // Handle ChromeCast player.
-        if (CastPlayer.castSession) {
+<c:if test="${model.googleCast}">
```
Comment (fxthomas) [inline airsonic-main/src/main/webapp/WEB-INF/jsp/playQueue.jsp]: The mixing of JSP and JS conditions is hard to read. How about doing something like setting the Google Cast instance to `null` at initialization if it's not enabled, and checking on that in JS only?

**Pull request closed without merging**